### ISSUE TITLE
False positive with while_let_on_iterator

### ISSUE THREAD

**Issue by sinkuu** [Manishearth/rust-clippy#1188]
Code:
```rust
 // Consume the whole comment
while let Some((FullCodeCharKind::InComment, (_, _))) = self.iter.next() {}
```

Clippy suggestion against this code:

```
warning: this loop could be written as a `for` loop, #[warn(while_let_on_iterator)] on by default
   --> src/comment.rs:419:17
    |
419 |                 while let Some((FullCodeCharKind::InComment, (_, _))) = self.iter.next() {}
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
help: try
    |                 for (FullCodeCharKind::InComment, (_, _)) in self.iter { .. }
    = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#while_let_on_iterator
```

These are not equivalent since `for` can't have refutable pattern.
